### PR TITLE
Merge for new attributes support in Build2017

### PR DESCRIPTION
**Pull request by sldi-microsoft**
Please do not merge this pull request until the BUILD release deployed.
Add new attributes contract 
Add ListPersons interface and deprecat GetPersons 
Support EnumString convert by Gson 
Display new attributes in Detection demo page 
Change version for Build2017

### CODE REVIEW

File: ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/FaceServiceClient.java [hunk 1]
```diff
@@ -329,15 +378,47 @@ public String toString() {
...
     /**
+     * List the fist "top" of persons whose Id is lager than "start".
+     * @param personGroupId The person group id.
+     * @param start The persons Id bar, list persons whose Id is lager than "start.
+     * @param top The number of persons to list.
+     * @return The person entity array.
+     * @throws ClientException
+     * @throws IOException
+     */
+    Person[] listPersons(String personGroupId, String start, int top) throws ClientException, IOException;
```
Comment (wangjun-microsoft) [inline ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/FaceServiceClient.java]: missing the function with only "top" parameter.

Comment (wangjun-microsoft) [inline ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/FaceServiceClient.java]: missing the function with only "top" parameter. Please check listPersongroups method as well.

Comment (sldi-microsoft) [inline ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/FaceServiceClient.java]: Fix it

Comment (sldi-microsoft) [inline ClientLibrary/lib/src/main/java/com/microsoft/projectoxford/face/FaceServiceClient.java]: Fix it

**Comment by sldi-microsoft** [comment]
Please do not merge this pull request until the BUILD release deployed.

File: Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java [hunk 2]
```diff
@@ -371,22 +385,77 @@ public View getView(final int position, View convertView, ViewGroup parent) {
...
+            else
+            {
+                String accessoriesType = "";
+
+                for (int i = 0; i < accessories.length - 1; ++i)
+                {
+                    accessoriesType = accessoriesType + accessories[i].type.toString() + " ";
+                }
+
+                accessoriesType = accessoriesType + accessories[accessories.length - 1].type.toString();
```
Comment (huxuan) [inline Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java]: Please use String.join [1] instead.

[1] https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.CharSequence...-

Comment (sldi-microsoft) [inline Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java]: Already fix it.

File: Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java [hunk 3]
```diff
@@ -68,7 +76,9 @@
 import java.text.DecimalFormat;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.LinkedHashSet;
```
Comment (huxuan) [inline Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java]: I do not think we need to use this kind of data structure.
Arrays [1] should be sufficient.

[1] https://developer.android.com/reference/java/util/Arrays.html

File: Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java [hunk 4]
```diff
@@ -68,7 +76,9 @@
...
 import java.util.List;
+import java.util.Set;
```
Comment (huxuan) [inline Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java]: Please remove it if it is not used in other places.

File: Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java [hunk 5]
```diff
@@ -443,15 +446,13 @@ private String getAccessories(Accessory[] accessories) {
             }
             else
             {
-                String accessoriesType = "";
-
-                for (int i = 0; i < accessories.length - 1; ++i)
+                Set<String> accessoriesSet = new LinkedHashSet<>();
```
Comment (huxuan) [inline Sample/app/src/main/java/com/microsoft/projectoxford/face/samples/ui/DetectionActivity.java]: Use `Array` to keep it simple.

**Pull request merged**